lib.sh

```diff
@@ -98,6 +99,55 @@ set_gac_filepath(){
     unset GAC_JSON;
 }
 
+# print a space-separated list of labels when run under Cirrus-CI for a PR
+get_pr_labels() {
+    req_env_vars CIRRUS_CI CIRRUS_PR CIRRUS_REPO_CLONE_TOKEN
+    req_env_vars CIRRUS_REPO_OWNER CIRRUS_REPO_NAME
+
+    local CURL query h_accept h_content api result filter labels
+    local h_auth h_accept h_content
+    # shellcheck disable=SC2154
+    h_auth="Authorization: bearer $CIRRUS_REPO_CLONE_TOKEN"
+    h_accept='Accept: application/vnd.github.antiope-preview+json'
+    h_content='Content-Type: application/json'
+    api="https://api.github.com/graphql"
+    # shellcheck disable=SC2154
+    query="{
+        \"query\": \"query {
+          repository(owner: \\\"$CIRRUS_REPO_OWNER\\\",
+                     name: \\\"$CIRRUS_REPO_NAME\\\") {
+            pullRequest(number: $CIRRUS_PR) {
+              labels(first: 100) {
+                nodes {
+                  name
+                }
+              }
+            }
+          }
+        }\"
+    }"
+    filter=".data.repository.pullRequest.labels.nodes[].name"
+    dbg "Issuing '$query'"
+    result=$(curl --location \
+             -H "$h_auth" -H "$h_accept" -H "$h_content" \
+             --request POST --data @- --url "$api" <<<"$query") \
+             || \
+             die "Error communicating with GraphQL API $api: $result"
+
+    # GraphQL sometimes returns errors inline, assume no result
+    # format or data should ever contain the string 'error'
+    if grep -q "error" <<<"$result"; then
+        die "Found 'error' in result: $result"
```

Comment:
Cringe: if anyone ever adds a label named 'ignore-errors' or 'terrorific', this will break in hard-to-diagnose ways. Could you check instead for an empty list in the `jq` below?